examples/subprotocol_server_youi/subprotocol_server_youi.cpp

```diff
@@ -0,0 +1,90 @@
+#include <iostream>
+#include <vector>
+#define ASIO_STANDALONE
+#include <websocketpp/config/asio_no_tls.hpp>
+#include <websocketpp/server.hpp>
+
+typedef websocketpp::server<websocketpp::config::asio> server;
+
+using websocketpp::connection_hdl;
+using websocketpp::lib::placeholders::_1;
+using websocketpp::lib::placeholders::_2;
+using websocketpp::lib::bind;
+using websocketpp::lib::ref;
+
+void print_protocols(std::vector<char*> const &supported_protocols)
+{
+    std::cout << "Supported sub-protocols: ";
+    
+    for (auto p: supported_protocols) {
+        std::cout << p << " ";
+    }
+    
+    std::cout << std::endl;
+}
+
+bool validate_protocol(std::string requested, std::vector<char*> const &supported_protocols)
+{
+    for (auto p: supported_protocols) {
+        if(requested == p){
+            return true;
+        }
+    }
+    return false;
+}
+
+bool validate(server & s, std::vector<char*> const &supported_protocols, connection_hdl hdl) {
+    server::connection_ptr con = s.get_con_from_hdl(hdl);
+
+    std::cout << "Cache-Control: " << con->get_request_header("Cache-Control") << std::endl;
+
+    const std::vector<std::string> & subp_requests = con->get_requested_subprotocols();
+    std::vector<std::string>::const_iterator it;
+
+    for (it = subp_requests.begin(); it != subp_requests.end(); ++it) {
+        std::cout << "Requested: " << *it << std::endl;
+        if(!validate_protocol(*it, supported_protocols))
+        {
+            std::cout << "Rejected" << std::endl;
+            print_protocols(supported_protocols);
+            return false;
+        }
+    }
+
+    if (subp_requests.size() > 0) {
+        con->select_subprotocol(subp_requests[0]);
+    }
+
+    return true;
+}
+
+void on_message(server &s, connection_hdl hdl, server::message_ptr msg)
+{
+    server::connection_ptr con = s.get_con_from_hdl(hdl);
+    
+    std::cout << "Received: " << msg->get_raw_payload() << std::endl;
+    
+    con->send(msg->get_raw_payload());
+    
+}
+
+int main(int argc, char *argv[]) {
+    try {
+        server s;
+        
+        std::vector<char*> supported_protocols(argv + 1, argv + argc);
```

Comment:
makes it more in line to what we do.
```suggestion
        std::vector<char*> supported_protocols(&argv[1], &argv[argc - 1]);
```